db/migrate/20230209194717_add_devise_to_users.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[7.0]
+  def self.up
+    change_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
```

Comment:
- [ ] Good job so far 👏 👏 I acknowledge how you remembered to add the `:confirmable` attribute for email validation, however, there is something that is occurring when I try to sign up. Here is the error that I am getting 👇 
![image](https://user-images.githubusercontent.com/90222110/217941389-a1c739ad-be57-4dad-be53-ba4387d93f28.png)

And what they are saying is understandable because when I look at your schema, I saw that among the columns of the `users` table, there is no `confirmed_at` 
![image](https://user-images.githubusercontent.com/90222110/217941711-b808d596-fc26-4a59-8642-a85664763b8a.png)

But here is the thing DON'T WORRY! It is easy to FIX 😃 because it is just one step that you forgot to do in order to finish devise configuration with `:confirmable` attribute.  In this migration, you were supposed to uncomment the codes under the header `# Confirmable` which is right above here just before you run it 😃 
so, the only thing I would suggest that you do is to edit it, by uncommenting what I said and then, `re-run` that specific migration. That's one of the workarounds you can use. Others may consider reverting the migration, edit the migration file and then run the migration again but at the end of the day, we are trying to update our `schema` and we can do this through migrations

Reply:
I am extremely grateful to you for your detail review. I did everything as you guided me.

Reply:
Screenshot of changes done 


![Screenshot (223)](https://user-images.githubusercontent.com/106825373/217991389-058c1cab-bd5c-4c6a-b8c5-a69be4998685.png)
![Screenshot (224)](https://user-images.githubusercontent.com/106825373/217991396-fae04989-0325-4f32-84b1-966b2846b7bb.png)